.github/workflows/build-and-deploy.yml

```diff
@@ -1,5 +1,7 @@
 name: Build and deploy
 
+permissions: read-all
```

Comment:
what does it do?

Reply:
It limits the permissions that the GITHUB_TOKEN will get when running workflows. By default it's set to allow write access.